tests/functional/u/unspecified_encoding_py38.py

```diff
@@ -81,3 +83,20 @@
 Path(FILENAME).open("w+")  # [unspecified-encoding]
 Path(FILENAME).open("w", encoding=None)  # [unspecified-encoding]
 Path(FILENAME).open("w", encoding=LOCALE_ENCODING)
+
+
+# Tests for storing data about the open call in a dataclass
+
+
+@dataclasses.dataclass
+class IOArgs:
+    """Dataclass storing information about how to open a file"""
+
+    encoding: Optional[str]
+    mode: str
+
+
+args_good_one = IOArgs(encoding=None, mode="wb")
+
+# Test for crash reported in https://github.com/PyCQA/pylint/issues/5321
+open(FILENAME, args_good_one.mode, encoding=args_good_one.encoding)
```

Comment:
Is it worth to infer the value of ``args_good_one.encoding`` to fix the false negative for unspecified encoding properly ?

Reply:
At the moment we can't (I think) as the issue with ``dataclass`` is still here. I used the tests from the issue as I thought it would be easy to add the other tests later on when that issue gets fixed!

Reply:
> Is it worth to infer the value of `args_good_one.encoding` to fix the false negative for unspecified encoding properly ?

I think you meant the `mode` argument. That will be inferred already. What's missing is just the check that the inferred value is actually a `Const` node and not something else. Only then can we access the `value` attribute. This is what this PR adds.
https://github.com/PyCQA/pylint/blob/2e7b2c20cb08e739dfcc83c8ae1508a96ee8e04d/pylint/checkers/stdlib.py#L627-L628